### PR TITLE
Refactor FXIOS-7673 [v122] Remove SnapKit from SearchViewController 

### PR DESCRIPTION
**Pull request by pelumy**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7673)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17109)

## :bulb: Description
- Removed `SnapKit` constraint on `SearchViewController` in the `BrowserViewController`
- Replaced `SnapKit` constraint with `NSLayoutConstraint`
- Replaced `SnapKit` constraint in `SearchViewController`  with `NSLayoutConstraint`

Video showing how it works after change:

https://github.com/mozilla-mobile/firefox-ios/assets/54324355/14753215-b28e-4bb3-8f0a-bd9ca67a2eee

iPad:
iPad View

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by lmarceau — COMMENTED** [review]
Hello @pelumy! Thank you for your work, this is great but we would like to remove as well the snapkit constraints from the `SearchViewController.swift` file if possible (if you search for `snp.` you'll see 10 matches in there). Do you want to look into those too? Can be done either with this PR or another one, as you wish!

**Comment by pelumy** [comment]
> Hello @pelumy! Thank you for your work, this is great but we would like to remove as well the snapkit constraints from the `SearchViewController.swift` file if possible (if you search for `snp.` you'll see 10 matches in there). Do you want to look into those too? Can be done either with this PR or another one, as you wish!

I will include the changes in this PR. Thanks for pointing it out.

**Review by lmarceau — CHANGES_REQUESTED** [review]
Hello! Couple of things:

With the latest changes, it seems there are some issues with top and bottom URL bar when showing the search view. We should be seeing three things with either top or bottom URL bar when we start typing for a search:
    1. The URL bar still need to be shown
    2. The search view controller shows the suggestions when we type
    3. The search engines suggestions bar should be appearing on top of the keyboard

I'm not seeing 1 and 3 when typing:
Simulator Screenshot - iPhone 15 Plus - 2023-11-23 at 12 00 40

I think this is due to how the constraints were changed in a couple of places. With SnapKit, constraints were **either** applied to top or bottom of certain elements depending on if we have the bottom or top URL bar. With the current PR, constraints are applied to top and bottom all the time, so I think this is where the problem lies. Let me know if there's any more details needed 🙏

**Comment by pelumy** [comment]
> Hello! Couple of things:
> 
> With the latest changes, it seems there are some issues with top and bottom URL bar when showing the search view. We should be seeing three things with either top or bottom URL bar when we start typing for a search: 1. The URL bar still need to be shown 2. The search view controller shows the suggestions when we type 3. The search engines suggestions bar should be appearing on top of the keyboard
> 
> I'm not seeing 1 and 3 when typing: Simulator Screenshot - iPhone 15 Plus - 2023-11-23 at 12 00 40
> 
> I think this is due to how the constraints were changed in a couple of places. With SnapKit, constraints were **either** applied to top or bottom of certain elements depending on if we have the bottom or top URL bar. With the current PR, constraints are applied to top and bottom all the time, so I think this is where the problem lies. Let me know if there's any more details needed 🙏

Thanks for pointing it out. I will fix this issue.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -1077,13 +1077,14 @@ class BrowserViewController: UIViewController,
...
-            let constraintTarget = isBottomSearchBar ? overKeyboardContainer.snp.top : view.snp.bottom
-            make.bottom.equalTo(constraintTarget)
-        }
+        NSLayoutConstraint.activate([
+            searchController.view.topAnchor.constraint(equalTo: header.bottomAnchor),
+            searchController.view.leftAnchor.constraint(equalTo: view.leftAnchor),
+            searchController.view.rightAnchor.constraint(equalTo: view.rightAnchor)
+        ])
+        searchController.view.bottomAnchor.constraint(equalTo: overKeyboardContainer.topAnchor).isActive = isBottomSearchBar
+        searchController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = !isBottomSearchBar
```
Comment (pelumy) [inline Client/Frontend/Browser/BrowserViewController.swift:1087]: @lmarceau , I can see that line 1086 and 1087 is where the issue is coming from. What I'm trying to figure out is why the search engine view disappears when I remove the previous line 1080 `searchController.view.snp.makeConstraints { make in }`. But with this line of code and the new constraints, the search engines show up.

**Comment by lmarceau** [comment]
Hello! Thanks @pelumy this great work! I pulled in the changes and there seems to be one thing left. With the top bar enabled, if I try to scroll on the search view controller the keyboard is being dismissed. The search view controller should then take the space so we can see the result full page, as it's done on `main`. I tested with bottom URL bar and all seemed good over there 👍 

Thank you, let me know if there's anything!

**Review by cyndichin — APPROVED** [review]
thanks for the changes! looks good on my side :)

**Review by lmarceau — APPROVED** [review]
Great work!! 🎉

**Pull request merged**